src/prelude/bytes.mare

```diff
@@ -503,3 +503,11 @@
       ._assign_at(0, number)
     @_size += U64.byte_width.usize
     @
+
+  :fun non join(others Array(Bytes), separator Bytes'val) Bytes'ref
+    res = Bytes.new
```

Comment:
With reference capabilities, it's a good practice to:

- :one: accept the "weakest" cap possible on your input that still lets you do the job
- :two: emit the "strongest" cap possible on your return value that you can manage to build efficiently

The combination of these two factors makes your function more usable to callers, since you're minimizing what powers you're asking them for, and maximizing what powers you give them.

In this case I think the best easy improvement to make is to buff up the return value by returning a `Bytes'iso`:

---

Also, it would be cool to have a default empty separator, so that joining a list of bytes with no separator is even easier. 

---

Both changes are reflected below. Note that adding a default value helped it to infer the parameter type, so I was able to omit the `Bytes'val` designation:

```suggestion
  :fun non join(others Array(Bytes), separator = b"") Bytes'iso
    res = Bytes.new_iso
```